src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -430,7 +430,8 @@ private void createActions(Bundle extras, NotificationCompat.Builder mBuilder, R
                 JSONArray actionsArray = new JSONArray(actions);
                 ArrayList<NotificationCompat.Action> wActions = new ArrayList<NotificationCompat.Action>();
                 for (int i=0; i < actionsArray.length(); i++) {
-                    int uniquePendingIntentRequestCode = i + notId;
+                    // This effectively appends i's value to the notId to ensure uniqueness when multiple notId's are sequential
+                    int uniquePendingIntentRequestCode = notId * 10 + i;
```

Comment:
@macdonst Yeah, I guess you could if your notId is greater than 214,748,364.  

I had to do it because I had a two notId's that were 100 and 101 so the codes were 100, 101, 102 and 101, 102, 103, respectively.  If I multiply by 10 with my change, they'll be 1000, 1001, 1002 and 1010, 1011, and 1012.   Note those in the second group are all unique. 

Should I just generate random integers like you originally suggested?